### DISCUSSION TITLE
Combobox virtualized

### DISCUSSION THREAD

**Discussion by ducan-ne** [heroui-inc/heroui#1867] [Help]
Hi, in the documentation, it is mentioned that the Combobox supports virtualized scrolling. I'm curious about how I can achieve that, is there any example?

https://nextui.org/docs/components/autocomplete#accessibility
> Support for virtualized scrolling for performance with long lists

**Comment by ducan-ne** [comment]
![CleanShot 2023-11-04 at 22 32 17@2x](https://github.com/nextui-org/nextui/assets/16853175/c4619920-5942-4e0b-a1e0-e52756e5aa40)
I discovered that there is the "isVirtualized" prop, but it seems like it is deprecated, and only tell react-aria that we're in a virtual scrolling listbox, without actually providing the functionality.

**Reply by awestm** [reply]
Did you find a resolution?

**Reply by BrainyWayne** [reply]
Is there a solution to this?

**Reply by ducan-ne** [reply]
Sorry but no

**Reply by awestm** [reply]
@BrainyWayne 

I actually did end up solving this. The autocomplete component supports async loading, you can follow the example on the docs: https://nextui.org/docs/components/autocomplete.

I won't say it's perfect and the example is a little difficult to follow, but essentially you can batch up a list of options into chunks of 50-100 and then load the next when the scroll limit is reached using `useInfiniteScroll`.

**Reply by BrainyWayne** [reply]
Awesome @awestm, trying this approach!

**Reply by LPSebi** [reply]
> @BrainyWayne
> 
> I actually did end up solving this. The autocomplete component supports async loading, you can follow the example on the docs: https://nextui.org/docs/components/autocomplete.
> 
> I won't say it's perfect and the example is a little difficult to follow, but essentially you can batch up a list of options into chunks of 50-100 and then load the next when the scroll limit is reached using `useInfiniteScroll`.

@awestm 
Would you mind posting the code or opening a PR to maybe add "isVirtualized"?

**Comment by alipalvane** [comment]
use https://tanstack.com/virtual/latest/docs/installation for support 
this text is from nextui document:
_Note: The virtualization strategy is based on the [@tanstack/react-virtual](https://tanstack.com/virtual/latest) package, which provides efficient rendering of large lists by only rendering items that are visible in the viewport._

**Comment by wingkwong** [comment]
see https://www.heroui.com/docs/components/autocomplete#virtualization
